dev/cuda/common.h

```diff
@@ -18,6 +18,25 @@ __device__ float warpReduceSum(float val) {
     return val;
 }
 
+// modified from: https://developer.nvidia.com/blog/faster-parallel-reductions-kepler/
+__inline__ __device__ float blockAllReduceSum(float val) {
```

Comment:
i think we already have this btw? see cuda_utils.cuh

Reply:
Yeah I found it there as well, just wasnt sure if it was a good idea to add it there too given that there is a warpReduceSum in both cuda_utils.h and common. I'll remove it.